plugins/msg-broker/mcollective/lib/openshift/mcollective_application_container_proxy.rb

```diff
@@ -3222,14 +3222,25 @@ def self.rpc_find_all_available(opts=nil)
               end
             end
 
+            # Ignore least_preferred_servers if all servers are considered least preferred
+            if least_preferred_servers.present?
+              ignore_least_preferred_servers = false
+              server_identities = server_infos.map {|server_info| server_info.name}.sort
+              if least_preferred_servers.compact.sort == server_identities
```

Comment:
You can't be sure that all servers are returned in server_infos for multiple reasons including:
- They won't be in the list if they are full
- They won't be in the list if the missed the mco discovery timeout.